src/common.rs

```diff
@@ -1891,6 +1891,278 @@ fn input_account_id_from_used_account_list(
+    fn from(predecessor_id: near_primitives::types::AccountId) -> Self {
+        Self::PredecessorId(predecessor_id)
+    }
+}
+
+impl From<near_crypto::PublicKey> for PermissionKey {
+    fn from(public_key: near_crypto::PublicKey) -> Self {
+        Self::PublicKey(public_key)
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+struct IsWritePermissionGrantedInputArgs {
+    key: String,
+    #[serde(flatten)]
+    permission_key: PermissionKey,
+}
+
+pub fn is_write_permission_granted<P: Into<PermissionKey>>(
+    network_config: &crate::config::NetworkConfig,
+    near_social_account_id: &near_primitives::types::AccountId,
+    permission_key: P,
+    key: String,
+) -> color_eyre::eyre::Result<bool> {
+    let function_args = serde_json::to_string(&IsWritePermissionGrantedInputArgs {
+        key,
+        permission_key: permission_key.into(),
+    })
+    .wrap_err("Internal error: could not serialize `is_write_permission_granted` input args")?;
+    let call_result = network_config
+        .json_rpc_client()
+        .blocking_call_view_function(
+            near_social_account_id,
+            "is_write_permission_granted",
+            function_args.into_bytes(),
+            near_primitives::types::Finality::Final.into(),
+        )
+        .wrap_err_with(|| "Failed to fetch query for view method: 'is_write_permission_granted'")?;
+
+    let serde_call_result: serde_json::Value = call_result.parse_result_from_json()?;
+    let result = serde_call_result.as_bool().expect("Unexpected response");
+    Ok(result)
+}
+
+pub fn is_signer_access_key_function_call_access_can_call_set_on_social_db_account(
+    near_social_account_id: &near_primitives::types::AccountId,
+    access_key_permission: &near_primitives::views::AccessKeyPermissionView,
+) -> color_eyre::eyre::Result<bool> {
+    if let near_primitives::views::AccessKeyPermissionView::FunctionCall {
+        allowance: _,
+        receiver_id,
+        method_names,
+    } = access_key_permission
+    {
+        Ok(receiver_id == &near_social_account_id.to_string()
+            && (method_names.contains(&"set".to_string()) || method_names.is_empty()))
+    } else {
+        Ok(false)
+    }
+}
+
+pub fn get_access_key_permission(
+    network_config: &crate::config::NetworkConfig,
+    account_id: &near_primitives::types::AccountId,
+    public_key: &near_crypto::PublicKey,
+) -> color_eyre::eyre::Result<near_primitives::views::AccessKeyPermissionView> {
+    let permission = network_config
+        .json_rpc_client()
+        .blocking_call_view_access_key(
+            account_id,
+            public_key,
+            near_primitives::types::Finality::Final.into(),
+        )
+        .wrap_err_with(|| format!("Failed to fetch query 'view access key' for <{public_key}>",))?
+        .access_key_view()?
+        .permission;
+    Ok(permission)
+}
+
+pub fn get_deposit(
+    network_config: &crate::config::NetworkConfig,
```

Comment:
Let's extract all the logic for interacting with socialdb from bos-cli-rs into a standalone crate (let's create https://github.com/bos-cli-rs/near-socialdb-client-rs), so we can reuse that code from near-cli-rs and from bos-cli-rs without copy-pasting it. `near-cli-rs::common` is not the right place to put all these functions.

Reply:
Done